libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,508 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file transport_secure_sockets.c
+ * @brief Implementation of transport interface API with Secure Sockets.
+ */
+
+/* Standard includes. */
+#include <stdlib.h>
+#include <string.h>
+
+/* TCP/IP abstraction includes. */
+#include "iot_secure_sockets.h"
+#include "transport_secure_sockets.h"
+
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Set up the context send, receive timeout.
+ *
+ * @param[in] tcpSocket the context to be set.
+ * @param[in] sendTimeoutMs Timeout for transport send.
+ * @param[in] recvTimeoutMs Timeout for transport recv.
+ *
+ * @return #SOCKETS_ERROR_NONE or error code from Secure Sockets.
+ */
+static int32_t transportTimeoutSetup( Socket_t tcpSocket,
+                                      uint32_t sendTimeoutMs,
+                                      uint32_t recvTimeoutMs );
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ * @param[in] pServerInfo Server connection info.
+ * @param[in] pSocketsConfig socket configs for the connection.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY,
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, #TRANSPORT_SOCKET_STATUS_API_ERROR,
+ *         #TRANSPORT_SOCKET_STATUS_DNS_FAILURE, #TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE on failure.
+ */
+static TransportSocketStatus_t establishConnect( NetworkContext_t * pNetworkContext,
+                                                 const ServerInfo_t * pServerInfo,
+                                                 const SocketsConfig_t * pSocketsConfig );
+
+/**
+ * @brief Set up TLS configurations for the socket.
+ *
+ * @param[in] pSocketsConfig the configure of session.
+ * @param[in] tcpSocket An initialized socket to secure.
+ * @param[in] pHostName Remote server name for SNI.
+ * @param[in] hostnameLength The length of `pHostName`.
+ *
+ * @return #SOCKETS_ERROR_NONE or #SOCKETS_SOCKET_ERROR.
+ */
+static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
+                         Socket_t tcpSocket,
+                         const uint8_t * pHostName,
+                         size_t hostnameLength );
+
+
+/*-----------------------------------------------------------*/
+
+int32_t SecureSocketsTrasnport_Send( NetworkContext_t * pNetworkContext,
```

Comment:
SecureSocketsTransport_Send

Reply:
fixed in 637fc71df6534bebe6497a06ed7e87c6fb7c8028